include/roboteam_ai/stp/Play.hpp

```diff
@@ -60,11 +61,19 @@ class Play {
     virtual void calculateInfoForRoles() noexcept = 0;
 
     /**
-     * Gets the score for the current play that is in the range of 0 - 255
-     * @param world World to get the score for
-     * @return The score, 0 - 255
+     * Calculate info for roles that are used in the scoring of the play.
+     * This is a purely virtual function, so it is implemented in every play.
+     */
+    virtual void calculateInfoForScoredRoles(world::World* world) noexcept = 0;
+
+    /**
+     * Function  for inbetween plays calculation of score through the PlayEvaluator.
+     * Using the struct PlayEvaluator::PlayScoring(uint8_t score, double weight) the factors the be considered
+     * can be defined for scoring a play.
+     * @param playEvaluator with the world
+     * @return a final score for the play
      */
-    [[nodiscard]] virtual uint8_t score(world::World* world) noexcept = 0;
+    virtual uint8_t score(PlayEvaluator *playEvaluator) noexcept = 0;
```

Comment:
shouldn't we pass these as ref? rather than pionter.

Reply:
We should, fixed it